datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -277,10 +287,15 @@ impl CaseExpr {
         // start with nulls as default output
         let mut current_value = new_null_array(&return_type, batch.num_rows());
         let mut remainder = BooleanArray::from(vec![true; batch.num_rows()]);
+        let mut remainder_count = batch.num_rows();
```

Comment:
given the similarity of this code and the one above, I wonder if there is some way to avoid the duplication (as part of a follow on PR)

Reply:
In #18152 the code is changing a bit further. If the approach there pans out I want to try to do the same for `case_when_with_expr`. It'll be easier to see if there's an extractable pattern once that work settles down, so if it's ok with you I would like to postpone your suggestion for a little bit.